drivers/ecg/adas1000/adas1000.c

```diff
@@ -0,0 +1,438 @@
+	case ADAS1000_31_25HZ_FRAME_RATE:
+		frm_ctrl_regval |= ADAS1000_FRMCTL_FRMRATE_31_25HZ;
+		break;
+	default: /** ADAS1000_2KHZ__FRAME_RATE */
+		frm_ctrl_regval |= ADAS1000_FRMCTL_FRMRATE_2KHZ;
+		break;
+	}
+
+	/** Write the new Frame control Register value */
+	return adas1000_write(device, ADAS1000_FRMCTL, frm_ctrl_regval);
+}
+
+/**
+ * @brief Reads the specified number of frames.
+ * @param device - Device structure.
+ * @param data_buff - Buffer to store the read data.
+ * @param frame_cnt - Number of frames to read.
+ * @param read_data_param - Structure holding the parameters required for frame
+			    read sequence
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t adas1000_read_data(struct adas1000_dev *device, uint8_t *data_buff,
+			   uint32_t frame_cnt, struct read_param *read_data_param)
+{
+	int32_t ret;
+	uint32_t data = 0;
+	uint32_t ready = 0;
+	uint32_t buff_size = 4;
+
+	/** If the read sequence must be started send a FRAMES command. */
+	if (read_data_param->start_read) {
+		ret = adas1000_write(device, ADAS1000_FRAMES, data);
+		if (ret != SUCCESS)
+			return ret;
+	}
+
+	/** Read the number of requested frames. */
+	while(frame_cnt) {
+		/** If waiting for the READY bit to be set read the header
+		until the bit is set, otherwise just read the entire frame. */
+		if(read_data_param->wait_for_ready) {
+			ready = 1;
+			while(ready == 1) {
+				/** if the header is repeated until the READY bit is set
+				read only the header, otherwise read the entire frame. */
+				if(read_data_param->ready_repeat) {
+					ret = spi_write_and_read(device->spi_desc,
+								 data_buff, buff_size);
+					if (ret != SUCCESS)
+						return ret;
+
+					ready = *data_buff & ADAS1000_RDY_MASK;
+					if(ready == 0) {
+						ret = spi_write_and_read(device->spi_desc,
+									 data_buff + 4,
+									 device->frame_size - 4);
+						if (ret != SUCCESS)
+							return ret;
+
+						data_buff += device->frame_size;
+						frame_cnt--;
+					}
+				} else {
+					ret = spi_write_and_read(device->spi_desc,
+								 data_buff, device->frame_size);
+					if (ret != SUCCESS)
+						return ret;
+
+					ready = *data_buff & ADAS1000_RDY_MASK;
+					if(ready == 0) {
+						data_buff += device->frame_size;
+						frame_cnt--;
+					}
+				}
+			}
+		} else {
+			ret = spi_write_and_read(device->spi_desc, data_buff, device->frame_size);
+			if (ret != SUCCESS)
+				return ret;
+
+			data_buff += device->frame_size;
+			frame_cnt--;
+		}
+	}
+
+	/** If the frames read sequence must be stopped read a register to stop the frames read. */
+	if(read_data_param->stop_read)
+		return adas1000_read(device, ADAS1000_FRMCTL, &ready);
+
+	return ret;
+}
+
+/**
+ * @brief Computes the CRC for a frame.
+ * @param device - Device structure.
+ * @param buff - Buffer holding the frame data.
+ * @return Returns the CRC value for the given frame.
+ */
+uint32_t adas1000_compute_frame_crc(struct adas1000_dev * device, uint8_t *buff)
```

Comment:
You can use: https://github.com/analogdevicesinc/no-OS/blob/master/include/crc16.h
and maybe will be useful to add crc24.h

Reply:
will add crc24 implementation, and adapt this driver to it.

Reply:
After https://github.com/analogdevicesinc/no-OS/pull/916 is merged will do the changes.